app/controllers/posts_controller.rb

```diff
@@ -7,4 +7,28 @@ def index
   def show
     @post = Post.find(params[:id])
   end
+
+  def new
+    @post = Post.new
+    @user = current_user
+  end
+
+  def create
+    @post = Post.new(post_params)
+    @post.author_id = current_user.id
+
+    respond_to do |format|
+      if @post.save
+        format.html { redirect_to user_post_url(current_user, @post), notice: 'Post was successfully created.' }
+      else
+        format.html { render :new, status: :unprocessable_entity }
+      end
+    end
+  end
+
+  private
+
+  def post_params
+    params.require(:post).permit(:title, :text, :comments_counter, :likes_counter)
+  end
```

Comment:
- [ ] The allowed parameters to create a post only include `title` and `text`. Let's fix that
You can learn more about the strong parameters in this [lesson](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/lessons/Controllers_lesson.md#:~:text=Pay%20special%20attention%20to%20the%20strong%20parameters%20description%20in%20Rails%20Guides)

Reply:
I fixed this one.